tonic/src/request.rs

```diff
@@ -1,4 +1,23 @@
 use crate::metadata::MetadataMap;
+use futures_core::Stream;
+
+/// A marker trait that should be implemented for all input messages.
+pub trait RequestMessage {}
```

Comment:
I don't think we need this marker trait since there is only ever one generic?

Reply:
You are right, it is not strictly necessary and a previous implementation did not have it. I decided to include it because:

- It sort of helps us constrain the other trait implementations. It just feels weird for IntoRequest or IntoStreamingRequest impl to exist for any type that is not an input to a rpc Method.
- I am toying around with an implementation that would let us call the streaming methods with iterators and having this bound sort of helps. Not sure if this will work of if it even makes sense.
- It makes code generation extremely simple, basically 1 line per input type.
- Triggered by the breakage in tonic-interop, I initially thought that it was going to be necessary to expose this marker for users to implement in certain scenarios, but that seems not to be necessary. I found the reason why interop is broken and has nothing to do with this.

None of these reasons hold much water though. In the end, however, this should be an internal implementation detail and as long as we are not changing the public api willy-nilly we should be able to iterate on the actual implementation.

Reply:
Yeah, those points make sense, my real motivation for removing it is that it's not actually required and just simplifies things. Happy either way.

Reply:
My 2¢ is that since the `Request` type itself is unconstrained, this shouldn't be added; if we _are_ going to constrain the `Message` type on `IntoRequest`, we should also constrain `Request`, and I'd consider adding that in a separate branch.

Reply:
Also, I think it's potentially very surprising to generate implementations of a `doc(hidden)` trait _in a user's codebase_ — I'd find it weird to run `cargo doc` on my project and see implementations of a trait that doesn't appear to exist.

Reply:
I removed this trait as suggested. IntoRequest for T is generated now instead, please let me know what you think.